gui/src/editor.rs

```diff
@@ -94,37 +98,32 @@ pub struct Editor {
 }
 
 impl Editor {
-    pub fn new(gen_config: GenerationConfig, map_config: MapConfig) -> Editor {
-        let init_gen_configs: HashMap<String, GenerationConfig> =
-            GenerationConfig::get_all_configs();
-        let init_map_configs: HashMap<String, MapConfig> = MapConfig::get_all_configs();
-
-        // TODO: its kinda stupid to initialize this as its literally re-initialized anyways
-        // when starting the first map generation. But i dont wanna bother adding an Option here as
-        // the generator also holds the initial empty map which is used for visualization.
-        let gen = Generator::new(&gen_config, &map_config, Seed::from_u64(0));
-
-        let mut visualize_debug_layers: HashMap<&'static str, bool> = HashMap::new();
-        for layer_name in gen.debug_layers.keys() {
-            visualize_debug_layers.insert(layer_name, true);
-        }
+    pub fn new() -> Editor {
+        let gen_configs =
+            load_configs_from_dir::<GenerationConfig, _>("../data/configs/gen").unwrap();
+        let map_configs = load_configs_from_dir::<MapConfig, _>("../data/configs/map").unwrap();
```

Comment:
when trying to call the crates from the project root via cargo run --bin <crate> using ../data/.. fails. I'd just make everything relative to the project root.